labelbox/data/serialization/ndjson/base.py

```diff
@@ -8,15 +8,17 @@
 
 class DataRow(BaseModel):
     id: str = None
+    global_key: str = None
 
-    @validator('id', pre=True, always=True)
-    def validate_id(cls, v):
-        if v is None:
-            raise ValueError(
-                "Data row ids are not set. Use `LabelGenerator.add_to_dataset`,or `Label.create_data_row`. "
-                "You can also manually assign the id for each `BaseData` object"
-            )
-        return v
+    @root_validator()
+    def must_set_one(cls, values):
+        if bool(values.get('id')) == bool(values.get('global_key')):
+            raise ValueError("Must set either id or global_key")
+        return values
+
+    class Config:
```

Comment:
This allows pydantic to deserialize a json into DataRow class using either the field name, or the alias of the field name as the key identifier.
Setting `alias_generator = camel_case` allows JSON that contains `globalKey` to map to `global_key` field, since it evaluates an alias of `global_key` field as `globalKey`. And then setting `allow_population_by_field_name = True` allows mapping of the keys using _both_ the field_name and the alias.

Reply:
I just found out that we have `labelbox.utils._CamelCaseMixin` to do this, would be good to use that instead of adding a nested Config class.

Reply:
oh good find!